src/org/etools/j1939_84/controllers/part11/Part11Step13Controller.java

```diff
@@ -53,33 +87,581 @@ public class Part11Step13Controller extends StepController {
+            // 6.11.13.7.b. List data received in a table using lifetime, stored 100 hr,
+            // active 100hr for columns, and categories for rows.
+            getListener().onResult(ghgTrackingModule.formatTrackingTable(Stream.concat(ghgTrackingLifetimePackets.stream(),
+                                                                                       ghgTrackingPackets.stream())
+                                                                               .collect(Collectors.toList())));
+        }
+
+        for (int pg : List.of(GHG_ACTIVE_100_HR, GHG_STORED_100_HR)) {
+            GenericPacket hybridPacketForPg = haveResponseWithPg(ghgTrackingPackets, pg);
+            if (hybridPacketForPg == null) {
+                // 6.11.13.8.a. For all MY2024+ engines, Fail each PG query where no response was received.
+                if (getEngineModelYear() >= 2024) {
+                    addFailure("6.2.17.14.a - No response was received from "
+                            + module.getModuleName() + "for PG "
+                            + pg);
+                }
+                // 6.11.13.8.b. For MY2022-23 engines, Warn each PG query, where no response was received
+                if (getEngineModelYear() >= 2022 && getEngineModelYear() <= 2023) {
+                    addWarning("6.11.13.8.b - No response was received from "
+                            + module.getModuleName() + "for PG "
+                            + pg);
+                }
+            } else {
+                hybridPacketForPg.getSpns()
+                           .forEach(spn -> {
+                               if (spn.getRawValue() > 0xFAFFL) {
+                                         // 6.11.13.8.c - Fail each PG query where any value received is greater than
+                                         // FAFFh.
+                                         addFailure("6.11.13.8.c - Bin value received is greater than 0xFAFF "
+                                           + module.getModuleName() + " returned "
+                                           + Arrays.toString(spn.getBytes()));
+                               }
+                               if (spn.getSlot().toValue(spn.getBytes()) > 0) {
+                                         // 6.11.13.8.d - Fail each active 100 hr array value that is greater than zero
+                                         addFailure("6.11.13.8.d - Active 100 hr array value received is greater than zero "
+                                           + module.getModuleName() + " returned "
+                                           + Arrays.toString(spn.getBytes()));
+                               }
+                                     // FIXME: the need to be implemented
+                                     // @Joe just need to implement these yet.
+                                     // 6.11.13.8.e - Fail if active 100 hrs engine hours < 600 seconds. (where
```

Comment:
All of these can be implemented without help from Joe.

Reply:
You are correct.  It is on the list for Task 5 and he wasn't in a hurry for it.  He is in a hurry for the Task 3 & Task 4 stuff.